pkg/arvo/sys/arvo.hoon

```diff
@@ -156,6 +245,549 @@
+    ::
+    ~%  %le  +>+>  ~
+    =|  $:  ::  run: list of worklists
+            ::  out: pending output
+            ::  gem: worklist metadata
+            ::  dud: propagate error
+            ::
+            run=(list plan)
+            out=(list ovum)
+            gem=germ
+            dud=(unit goof)
+        ==
+    ::
+    |_  $:  our=ship
+            now=@da
+            eny=@uvJ
+            lac=?
+            vil=vile
+            van=(map term vane)
+        ==
+    +*  this  .
+    ::  +abet: finalize loop
+    ::
+    ++  abet
+      ^-  (pair (list ovum) (list (pair term vane)))
+      :-  (flop out)
+      %+  sort
+        ~(tap by van)
+      |=([[a=@tas *] [b=@tas *]] (aor a b))
+    ::  +emit: enqueue a worklist with source
+    ::
+    ++  emit
+      |=  [src=term moz=(list move)]
+      =/  =plan  [[src +(depth.gem)] moz]
+      this(run [plan run])
+    ::  +poke: prepare a worklist-of-one from outside
+    ::
+    ++  poke
+      |=  [vane=term =ovum]
+      ^+  this
+      ~?  !lac  ["" %unix p.card.ovum wire.ovum now]
+      =/  =maze
+        =/  =type  [%cell [%atom %tas `%soft] %noun]
+        [%& type [%soft card.ovum]]
+      =/  =move
+        ~|  [%poke %bad-wire wire.ovum]
+        ?>  ?=([%$ *] wire.ovum)
+        [duct=~ %pass t.wire.ovum vane maze]
+      (emit %$ move ~)
+    ::  +crud: prepare a worklist-of-one with error report from outside
+    ::
+    ++  crud
+      |=  [vane=term =goof =ovum]
+      ^+  this
+      ~?  !lac  ["" %unix %crud p.card.ovum wire.ovum now]
+      =/  =maze
+        =/  =type  [%cell [%atom %tas `%soft] %noun]
+        [%& type [%soft card.ovum]]
+      =/  =move
+        ~|  [%crud %bad-wire wire.ovum]
+        ?>  ?=([%$ *] wire.ovum)
+        [duct=~ %hurl goof %pass t.wire.ovum vane maze]
+      (emit %$ move ~)
+    ::  +spam: prepare a worklist for all targets
+    ::
+    ++  spam
+      |=  =ovum
+      ^+  this
+      =/  ord=(list (pair term *))
+        %+  sort
+          ~(tap by van)
+        |=([[a=@ *] [b=@ *]] (aor b a))
+      |-  ^+  this
+      ?~  ord
+        this
+      =.  this  (poke p.i.ord ovum)
+      $(ord t.ord)
+    ::  +loop: until done
+    ::
+    ++  loop
+      ^+  this
+      ?~  run
+        this
+      ?:  =(~ q.i.run)    :: XX TMI
+        loop(run t.run)
+      =.  dud  ~
+      =.  gem  p.i.run
+      =^  mov  q.i.run  q.i.run
+      loop:(step mov)
+    ::  +step: advance the loop one step by routing a move
+    ::
+    ++  step
+      |=  =move
+      ^+  this
+      ::
+      ~?  &(!lac ?=(^ dud))  %goof
+      ::
+      ?-  -.ball.move
+      ::
+      ::  %pass: forward move
+      ::
+          %pass
+        =*  wire  wire.ball.move
+        =*  duct  duct.move
+        =*  vane  vane.note.ball.move
+        =*  task  task.note.ball.move
+        ::
+        ~?  &(!lac !=(%$ vane.gem))
+          :-  (runt [(dec depth.gem) '|'] "")
+          :^  %pass  [vane.gem vane]
+            ?:  ?=(?(%deal %deal-gall) +>-.task)
```

Comment:
Why is %deal-gall special-cased here?  If it's because the move contains a vase, we might want to check for Ford builds here too.

Reply:
This is just ported straight from `+jack:is`, preserving the current behavior of `|verb`, which  has been documented, and is expected by some (small number of) people. I think `|verb` could be significantly improved, and also think we should probably have short and long printing routines in %zuse for all vanes' tasks/gifts.